drivers/spi/spi_nrfx_spim.c

```diff
@@ -59,7 +59,8 @@ static inline nrf_spim_frequency_t get_nrf_spim_frequency(uint32_t frequency)
 		return NRF_SPIM_FREQ_2M;
 	} else if (frequency < 8000000) {
 		return NRF_SPIM_FREQ_4M;
-#if defined(CONFIG_SOC_NRF52833) || defined(CONFIG_SOC_NRF52840)
+#if defined(CONFIG_SOC_NRF52833) || defined(CONFIG_SOC_NRF52840) \
+                                 || defined(CONFIG_SOC_NRF5340_CPUAPP)
```

Comment:
What about changing it to:
```
#if defined(SPIM_FREQUENCY_FREQUENCY_M32)
```
so that there is no need to update it again when another SoC with HS SPIM appears (see how `NRF_SPIM_FREQ_32M` is [defined](https://github.com/zephyrproject-rtos/hal_nordic/blob/3a1ee83b2d6fedfbbf73d81389c2d06b61a13041/nrfx/hal/nrf_spim.h#L148))?

Reply:
Confirmed that ```SPIM_FREQUENCY_FREQUENCY_M32``` is defined in following files:
* nrf52840_bitfields.h
* nrf52833_bitfields.h
* nrf5340_bitfields.h
So it should work, but not sure if we use ```#if defined(SPIM_FREQUENCY_FREQUENCY_M32)``` would make people confused?